nuxeo-ai-model/src/main/java/org/nuxeo/ai/bulk/DataSetBulkAction.java

```diff
@@ -127,41 +128,56 @@ public ExportingComputation(String name) {
 
         @Override
         protected void compute(CoreSession coreSession, List<String> ids, Map<String, Serializable> properties) {
+            log.debug("{} ids to export.", ids.size());
+
             String[] props = split((String) properties.get(EXPORT_FEATURES_PARAM), ",");
             String split = (String) properties.getOrDefault(EXPORT_SPLIT_PARAM, String.valueOf(DEFAULT_SPLIT));
 
             int percentSplit = Integer.parseInt(split);
-            ThreadLocalRandom random = ThreadLocalRandom.current();
-            for (DocumentModel doc : loadDocuments(coreSession, ids)) {
-                List<String> targetProperties = new ArrayList<>(asList(props));
-
-                if (doc.hasFacet(ENRICHMENT_FACET)) {
-                    SuggestionMetadataWrapper wrapper = new SuggestionMetadataWrapper(doc);
-                    targetProperties.removeAll(wrapper.getAutoFilled());
-                    targetProperties.removeAll(wrapper.getAutoCorrected());
-                }
 
-                BlobTextFromDocument subDoc = null;
-                if (!targetProperties.isEmpty()) {
-                    subDoc = PropertyUtils.docSerialize(doc, new HashSet<>(targetProperties));
-                }
+            // Split documents list into training and validation datasets
+            DocumentModelList docs = loadDocuments(coreSession, ids);
+            List<List<DocumentModel>> datasets = ListUtils.partition(docs, (docs.size() - 1) * percentSplit / 100);
```

Comment:
This will not work. We need a uniform sampling over time. If we split 20 from the beginning or end could bring evaluation errors later

Reply:
How could it work before so?

Reply:
As talked, shuffle all before a data cut